### DISCUSSION TITLE
Moddable Documentation Japanese Translation

### DISCUSSION THREAD

**Discussion by meganetaaan** [Moddable-OpenSource/moddable#1332] [Ideas]
Hello Moddable community,

We are [a group of Japanese developers who are passionate about the Moddable SDK](https://discord.gg/MnkTyXjK) and would like to contribute to its growth in Japan. We believe that Moddable deserves more recognition and adoption in our country.

One of the main challenges we have identified in promoting Moddable domestically is the lack of comprehensive Japanese language documentation. While there are some great articles and blog posts by Japanese developers, we feel that having an official Japanese translation of the Moddable documentation would be immensely beneficial.

I recently had the opportunity to discuss this idea with Hoddie from the Moddable team, and we were encouraged by his positive response. As a first step, we would like to propose a community-driven effort to translate the official Moddable documentation into Japanese.

Our initial plan is to:

- Fix the directory structure of the documentation
- Create a list of the current markdown documents and assign priority levels
- Translate the documents in order of priority

We are also considering using machine translation tools like ChatGPT to assist with the translation process, while ensuring that the final output is of high quality and human-reviewed.

Furthermore, we would like to discuss how to integrate the Japanese translation into [the official Moddable website](https://www.moddable.com/documentation/readme.php), making it easily accessible to Japanese users.

Through this initiative, we aim to make Moddable more accessible to Japanese developers and increase the number of users and contributors from Japan.

We would love to hear your thoughts, suggestions, and support for this project. Let's work together to make Moddable even more global and inclusive!

**Comment by phoddie** [comment]
This sounds great!

> Furthermore, we would like to discuss how to integrate the Japanese translation into [the official Moddable website](https://www.moddable.com/documentation/readme.php), making it easily accessible to Japanese users.

We have a tool that generates the HTML documentation from the markdown. We should be able to update this to support other languages.

> Fix the directory structure of the documentation

"Fix" has several meanings in English. I believe what you mean is that "we should decide where in the documentation structure to put the Japanese translations." Is that correct?

**Reply by meganetaaan** [reply]
>We have a tool that generates the HTML documentation from the markdown. We should be able to update this to support other languages.

That's great to hear! We appreciate your willingness to update the tool to support other languages. We look forward to collaborating with you on this.

>"Fix" has several meanings in English. I believe what you mean is that "we should decide where in the documentation structure to put the Japanese translations." Is that correct?

Thank you for your clarification. You're right about the directory structure.
Upon further reflection, I realized that we should take a step back and consider a broader question before discussing the specific location of the Japanese translations.
The main issue we need to address first is whether to use the same repository as Moddable or create a separate repository for the translations.

We have two main options:

1. Use the same repository as Moddable
2. Create a separate repository for Japanese translations

If we choose option 2, we can further consider:

2.1 Placing the repository within the Moddable organization
2.2 Placing the repository in another organization

Each approach has its own pros and cons. Using the same repository as Moddable would make it easier to maintain consistency and synchronization between the English and Japanese versions. It would also make the documentation more discoverable for users, as it would be located within the official Moddable repository.

On the other hand, creating a separate repository would allow the Japanese community to work more autonomously on translations. If the repository is placed within the Moddable organization (option 2.1), it would still be closely associated with the official Moddable project.

Please share your thoughts on these options. We can further discuss the pros and cons of each approach and decide on the best way forward.

**Reply by phoddie** [reply]
I think I prefer your option 2.1.

- In the "Moddable-OpenSource" organization. This makes if more official, and let's us use our usual CLA to ensure everyone is free to use the result
- Separate repository. This allows for frequented updates by a larger group of contributors.

It would be good if the repository follows the structure of the Moddable SDK. That way, any document in any location (such as readme.md documentation for examples) can possibly be translated. It eliminates the need to provide any kind of separate mapping  from the translated documentation to the documentation in the Moddable SDK.

We **might** merge the translations Ito the Moddable SDK directory periodically. I'm not sure. That is something we can investigate together and decide later.

**Reply by meganetaaan** [reply]
Agreed! The same structure as Moddable SDK is pretty straightforward.

I think we have reached a consensus on the basic approach, so we can start writing. Would you please create a new repository for the Japanese development community to begin working on the translations?

**Reply by phoddie** [reply]
Sure. There is now a public [moddable-jp](https://github.com/Moddable-OpenSource/moddable-jp) repository. We can change the name, if there is a better idea. You are a maintainer for the repository.

**Reply by phoddie** [reply]
I recently noticed the great progress of the translation effort. 🎉 Congratulations! 🎉 Your work has also contributed to improving the English documentation.

I thought it might be time to think about how we will publish the translation. The goal should be that developers working in Japanese can access the translations effortlessly. There are two places where we publish the documentation:

- The Moddable SDK repository
- The [documentation section](https://www.moddable.com/documentation/readme) of the Moddable website

The Moddable website is much faster and easier to browse than the Moddable SDK repository. But, it does not contain all the documentation – only the contents of `$MODDABLE/documentation`. 

Has the translation team thought further about how you'd like to see the translations published for developers?
